src/pcp/types.rs

```diff
@@ -217,6 +217,133 @@ impl<F: FieldElement> Value for Sum<F> {
     }
 }
 
+/// The histogram type. Each measurement is a non-negative integer and the aggregate is a histogram
+/// approximating the distribution of the measurements.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Histogram<F> {
+    data: Vec<F>,
+    range_checker: Vec<F>,
+    sum_check_share: F,
+}
+
+impl<F: FieldElement> Histogram<F> {
+    /// Constructs a new histogram input. The values of `buckets` must be strictly increasing.
+    pub fn new(measurement: u64, buckets: &[u64]) -> Result<Self, PcpError> {
```

Comment:
I wonder if we could make `Histogram` generic over integer primitive types, and then allow construction of histograms with anything from `u32` to `u128`, provided that the chosen `FieldElement` is big enough to represent the chosen primitive integer? [trait `Unsigned`](https://docs.rs/num/0.4.0/num/trait.Unsigned.html) could work.

Reply:
`F` already provides `F::Integer`, which should be suitable here. It's slightly annoying to make the caller do the conversion from `u64` (or whatever) to `F::Integer`, but I don't it's too annoying. In any case, it probably makes sense at this (relatively low) level of the library.

If we do this, I think we should also do it for `Count` and `Sum`. Would you mind if I make a note to implement this change later, after everything else is done? (I've got a running list of TODOs.)

Reply:
> Would you mind if I make a note to implement this change later, after everything else is done? (I've got a running list of TODOs.)
Sounds good,